### ISSUE TITLE
Inconsistency in updating test case status in ADO

### ISSUE THREAD

**Issue by mechandan** [alexneo2003/playwright-azure-reporter#42]
@alexneo2003  I can make sure that there is something wrong and requesting you to have a look .

i have tried many times and able to reproduce this issue.

Note :- All the test cases Ids are different 
            Either its testRun / testResults , I am able to reproduce in both the scenarios

Seems like there is a bug where if the test case once fail / blocked n in next run its passes , the status of the test case remains fail / block though in actual it is pass

steps to reproduce :

Execute the test which fail in execution.
Test status in ADO is fail ( thats fine )
Fix the test case and run it again , this time test is passing
Expected Result :- the failed test case status should change from fail to pass

Actual result : Remains Fail

**Comment by alexneo2003** [comment]
First two (from bottom) was runned with `publishTestResultsMode: 'testResult'` config
second two (from bottom) was runned with `publishTestResultsMode: 'testRun'` config

<img width="1074" alt="Screenshot 2023-09-12 at 13 28 16" src="https://github.com/alexneo2003/playwright-azure-reporter/assets/11705902/e428af8d-911a-43a7-bc95-32216c172663">

<img width="333" alt="Screenshot 2023-09-12 at 13 27 53" src="https://github.com/alexneo2003/playwright-azure-reporter/assets/11705902/21a367b5-c49a-47ba-a251-1b7c3795c4ed">

as you can see TC status was changed from `failed` to `success`

**Comment by mechandan** [comment]
> First two (from bottom) was runned with `publishTestResultsMode: 'testResult'` config second two (from bottom) was runned with `publishTestResultsMode: 'testRun'` config
> 
> <img alt="Screenshot 2023-09-12 at 13 28 16" width="1074" src="https://user-images.githubusercontent.com/11705902/267299166-e428af8d-911a-43a7-bc95-32216c172663.png"> <img alt="Screenshot 2023-09-12 at 13 27 53" width="333" src="https://user-images.githubusercontent.com/11705902/267299646-21a367b5-c49a-47ba-a251-1b7c3795c4ed.png">
> as you can see TC status was changed from `failed` to `success`


Ya but its not consistent @alexneo2003  I am using it since 1 week and can say this , could you please try to execute it multiple times n see .


n would you mine to share the API and the way you are updating the status please .
Thanks

**Comment by alexneo2003** [comment]
What api request?) All updates performed from this reporter

**Comment by mechandan** [comment]
I means to say in the reporter itself you might be doing some operation  to update the status of the test cases  right ? 
Const patchDocument = [
op:?
Path:?
values :?
]

**Comment by alexneo2003** [comment]
I mean that I wrote sample test with TC Id of existing in ADO TC. 
I use code like `expect(true).toBe(true)` for successful test and `expect(true).toBe(false)` for failed test. Run it one by one and check result in ADO.
I don't used any additional request from reporter or another cheating to update status

**Comment by mechandan** [comment]
Ok let me rephrase my question,

i am asking about ADO api that your are using in the reporter  to connect and update the status of a test case in test plan of the azure 


I am interested in the logic that you have written in your reporter

**Comment by alexneo2003** [comment]
At this moment you write message at issue in repository which store source code of reporter 🤔 you can explore it as long as you can

**Comment by mechandan** [comment]
Sure, please keep this issue open , I will try few more times , Thanks

**Comment by alexneo2003** [comment]
any updates?
